source/game/field/obj/ObjectKinoko.cc

```diff
@@ -0,0 +1,116 @@
+#include "game/field/obj/ObjectKinoko.hh"
+
+namespace Field {
+
+ObjectKinokoNm::ObjectKinokoNm(const System::MapdataGeoObj &params) : ObjectKCL(params) {
+    m_type = static_cast<KinokoType>(params.setting(0));
+}
+
+/// @addr{0x80827A9C}
+ObjectKinokoNm::~ObjectKinokoNm() = default;
+
+/// @addr{0x8080761C}
+ObjectKinoko::ObjectKinoko(const System::MapdataGeoObj &params)
+    : ObjectKCL(params), m_objPos(m_pos), m_objRot(m_rot) {
+    m_type = static_cast<KinokoType>(params.setting(0));
+
+    _12e = 0;
+    _144 = 6;
+    _130 = 40;
+    _132 = 10;
+    _12c = 1 % 40;
+    _134 = 0.1f;
+    m_period = 6.0f * F_PI / 40.0f;
+    _13c = 0.0008f;
```

Comment:
Let's define some local constexpr variables here:

- 40 frames is the pulse duration
- 10 frames is the rest duration
- `_12c` (pulse frame) is 1 % pulse duration
- `m_period` is `6.0f * F_PI` / pulse duration

We can remove `_13c` from the member list and instead declare it as a static constexpr f32 called `PULSE_SCALE`, and simply have `ObjectKinoko` reference it, as it won't change.

We should also maybe add a comment here to signal that `m_period` is initialized with 6pi so it will pulse three times for the duration.

Reply:
I know `1 % 40` is as a result of the holder index not being used properly, but can we reduce further and just hard-code it as 1, since `1 % n = 1` for `n >= 1`. This would also mean the `m_scale` assignment gets reduced to:
`m_scale.set(_134 * EGG::Mathf::sin(m_period) + 1.0f);`

Reply:
The `m_scale` assignment can't decay like this.